### DISCUSSION TITLE
compile CSS fill

### DISCUSSION THREAD

**Discussion by odaihatemali** [TDRA-ae/aegov-dls#28] [Q&A]
Dears, 

I'm trying to compile a CSS file like the documentation, but the output CSS file does not have all the components' styles, like cards, tabs, etc. 

Thanks,
Regards,

**Comment by moiz-conceptlz** [comment]
Hello @odaihatemali 

Let us first make sure that you have all the dependencies checked.

1. Have you added the path to your working files in the tailwind.config.js under "**content**". Is this pointing to your frontend and JS files?
2. Have you added the code of any component to your HTML or working files?

Tailwind CSS works by scanning all of your HTML files, JavaScript components, and any other templates for class names, generating the corresponding styles and then writing them to a static CSS file.

This ensures your rendered CSS file only includes the classes that are required. The underlying method of how this is done is via 'purgeCSS'

It is strongly recommended to use a build process rather than the command line interface. The CLI command is great for testing, but ideally you need to set up a build process that you are comfortable with - using webpack, gulp or just about anything that can process your build. You can also visit [https://tailwindcss.com/docs/installation/framework-guides](https://tailwindcss.com/docs/installation/framework-guides) for more details on some specific build processes for frameworks.

**Reply by odaihatemali** [reply]
@moiz-conceptlz 

Thanks, Moiz, for your quick response; I do precisely like documentation https://designsystem.gov.ae/docs/installation 

What am I doing, bro? I'm working on the Sharepoint portal, and I need all component styles in one file. How can I compile this without using any framework?

**Reply by moiz-conceptlz** [reply]
Hi @odaihatemali 

Even if you are using Sharepoint - your frontend code, your HTML, your Javascript is sitting somewhere. The way Tailwind CSS works is that it scans your code, and only uses the classes required to generate the CSS. 

It also highly depends on what you are using as your frontend stack. Are you using React / Angular or something like NextJS? If so, would already have a build system via npm. 

Assuming you are not using any of the frontend stacks, and you are coding your interface in pure HTML. The first thing you have to do -is to edit the **tailwind.config.js** file and point the `content` attribute to the folder (or sub folders using the ** wildcard) where your frontend files are. 

Now, here are some old (but still very useful) articles that talk about how to integrate and create a build system with Tailwind CSS and Sharepoint.

- [https://pnp.github.io/blog/post/how-to-set-up-tailwind-css-in-a-spfx-project/](https://pnp.github.io/blog/post/how-to-set-up-tailwind-css-in-a-spfx-project/)
- [https://adoption.microsoft.com/en-us/sample-solution-gallery/sample/pnp-sp-dev-spfx-web-parts-react-tailwindcss3/](https://adoption.microsoft.com/en-us/sample-solution-gallery/sample/pnp-sp-dev-spfx-web-parts-react-tailwindcss3/)
- Perhaps this video can help too - [https://www.youtube.com/watch?v=_gQkauERWh0](https://www.youtube.com/watch?v=_gQkauERWh0) 

The purpose of the design system is to offer a bare bone framework that can be used with pure HTML/JS. You are free to choose any method to integrate it with your CMS of choice.
